internal/services/discovery/openshift_test.go

```diff
@@ -0,0 +1,90 @@
+package discovery
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/yaml"
+	fakedynamic "k8s.io/client-go/dynamic/fake"
+	fakeclientset "k8s.io/client-go/kubernetes/fake"
+
+	"castai-agent/internal/services/controller/scheme"
+)
+
+func TestServiceImpl_GetOpenshiftClusterID(t *testing.T) {
+	r := require.New(t)
+
+	clusterVersionJSON := `
+apiVersion: config.openshift.io/v1
+kind: ClusterVersion
+metadata:
+  name: version
+  uid: dc0570f9-2c46-40b1-a5d0-c7233e82c7b6
+spec:
+  clusterID: 91d87440-5173-47f0-aca5-65bc0144ad30
+`
+
+	var clusterVersionMap map[string]any
+	r.NoError(yaml.Unmarshal([]byte(clusterVersionJSON), &clusterVersionMap))
+
+	version := &unstructured.Unstructured{}
+	version.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   OpenshiftClusterVersionsGVR.Group,
+		Version: OpenshiftClusterVersionsGVR.Version,
+		Kind:    "ClusterVersion",
+	})
+	version.Object = clusterVersionMap
```

Comment:
would be nice to have a helper string-to-object, e.g.

```suggestion
	version.Object = toYaml(r, `
apiVersion: config.openshift.io/v1
kind: ClusterVersion
metadata:
  name: version
  uid: dc0570f9-2c46-40b1-a5d0-c7233e82c7b6
spec:
  clusterID: 91d87440-5173-47f0-aca5-65bc0144ad30
`)

	... 	
func toYaml(r *require.Assertions, input YAML) map[string] any {
  	var res map[string]any
	r.NoError(yaml.Unmarshal([]byte(inputYAML), &res))
	return res
}
```